src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ML.Data
+{
+    public sealed partial class TextLoaderRange
+    {
+        public TextLoaderRange()
+        {
+        }
+
+        public TextLoaderRange(int ordinal)
+        {
+            Min = ordinal;
+            Max = ordinal;
+        }
+
+        public TextLoaderRange(int min, int max)
+        {
+            Min = min;
+            Max = max;
+        }
+    }
+
+    public sealed partial class TextLoader
+    {
+        /// <summary>
+        /// Construct a TextLoader object by inferencing the dataset schema from a type.
+        /// </summary>
+        /// <param name="useHeader">Does the file contains header?</param>
+        /// <param name="separator">Column separator character. Default is '\t'</param>
+        /// <param name="allowQuotedStrings">Whether the input may include quoted values, 
+        /// which can contain separator characters, colons,
+        /// and distinguish empty values from missing values. When true, consecutive separators 
+        /// denote a missing value and an empty value is denoted by \"\". 
+        /// When false, consecutive separators denote an empty value.</param>
+        /// <param name="supportSparse">Whether the input may include sparse representations e.g. 
+        /// if one of the row contains "5 2:6 4:3" that's mean there are 5 columns all zero 
+        /// except for 3rd and 5th columns which have values 6 and 3</param>
+        /// <param name="trimWhitespace">Remove trailing whitespace from lines</param>
+        public TextLoader CreateFrom<TInput>(bool useHeader = false,
+            char separator = '\t', bool allowQuotedStrings = true,
+            bool supportSparse = true, bool trimWhitespace = false)
+        {
+            var fields = typeof(TInput).GetFields();
+            Arguments.Column = new TextLoaderColumn[fields.Length];
+            for (int index = 0; index < fields.Length; index++)
+            {
+                var field = fields[index];
+                var mappingAttr = field.GetCustomAttribute<ColumnAttribute>();
+                if (mappingAttr == null)
```

Comment:
>mappingAttr [](start = 20, length = 11)

Do we want to throw here, or maybe just ignore?

Reply:
Can you please clarify?

---
In reply to: [188731983](https://github.com/dotnet/machinelearning/pull/142#discussion_r188731983) [](ancestors = 188731983)

Reply:
Ignoring doesn't quite work since you need to specify the ranges.

---
In reply to: [188844047](https://github.com/dotnet/machinelearning/pull/142#discussion_r188844047) [](ancestors = 188844047,188731983)

Reply:
I was just wondering whether there could be a case where a class has additional fields that don't have the ColumnAttribute and that would be ok.

---
In reply to: [189081999](https://github.com/dotnet/machinelearning/pull/142#discussion_r189081999) [](ancestors = 189081999,188844047,188731983)